app/models/commit_monitor_branch.rb

```diff
@@ -39,6 +39,14 @@ def pr_number
     MiqToolsServices::MiniGit.pr_number(name) if pull_request?
   end
 
+  def github_pr_uri
+    unless pull_request?
+      raise ArgumentError, "Cannot retrieve pull request URI for non-pull request " \
```

Comment:
I'm all for changing the error or returning nil, ArgumentError doesn't make much sense here.  Any suggestions for a different exception type?

I do think that this method makes more sense as an instance method because of the dependencies on object data.  It would be a bit cumbersome to pull that data out of an instance, then pass it back in to the class method.

The github_commit_uri method is a bit strange in that I think it could make sense as an instance method, but the only place it is being used is in the creation of the branch itself here:
https://github.com/ManageIQ/miq_bot/blob/master/app/models/commit_monitor_repo.rb#L20
I think it would be a bit odd to instantiate a branch object just to use that method to create a branch object.